src/storm/solver/TopologicalMinMaxLinearEquationSolver.cpp

```diff
@@ -240,36 +271,53 @@ namespace storm {
         
         template<typename ValueType>
         bool TopologicalMinMaxLinearEquationSolver<ValueType>::solveScc(storm::Environment const& sccSolverEnvironment, OptimizationDirection dir, storm::storage::BitVector const& sccRowGroups, storm::storage::BitVector const& sccRows, std::vector<ValueType>& globalX, std::vector<ValueType> const& globalB) const {
-
             // Set up the SCC solver
             if (!this->sccSolver) {
                 this->sccSolver = GeneralMinMaxLinearEquationSolverFactory<ValueType>().create(sccSolverEnvironment);
                 this->sccSolver->setCachingEnabled(true);
             }
-            if (this->choiceFixedForState) {
-                // convert fixed states to only fixed states of sccs
+            this->sccSolver->setHasUniqueSolution(this->hasUniqueSolution());
+            this->sccSolver->setHasNoEndComponents(this->hasNoEndComponents());
+            this->sccSolver->setTrackScheduler(this->isTrackSchedulerSet());
+
+            storm::storage::SparseMatrix<ValueType> sccA;
+            if (this->choiceFixedForRowGroup) {
+                // Obtain choiceFixedForState bitvector containing only the states of the scc.
                 storm::storage::BitVector choiceFixedForStateSCC(sccRowGroups.getNumberOfSetBits());
                 auto j = 0;
                 for (auto i : sccRowGroups) {
-                    choiceFixedForStateSCC.set(j, this->choiceFixedForState.get()[i]);
+                    choiceFixedForStateSCC.set(j, this->choiceFixedForRowGroup.get()[i]);
                     j++;
                 }
                 assert (j = sccRowGroups.getNumberOfSetBits());
-                this->sccSolver->setChoiceFixedForStates(std::move(choiceFixedForStateSCC));
-            }
-            this->sccSolver->setHasUniqueSolution(this->hasUniqueSolution());
-            this->sccSolver->setHasNoEndComponents(this->hasNoEndComponents());
-            this->sccSolver->setTrackScheduler(this->isTrackSchedulerSet());
-            
-            // SCC Matrix
-            storm::storage::SparseMatrix<ValueType> sccA;
-            if (this->choiceFixedForState) {
                 sccA = this->A->getSubmatrix(false, sccRows, sccRowGroups);
+
+                // initial scheduler
+                if (this->hasInitialScheduler()) {
+                    auto sccInitChoices = storm::utility::vector::filterVector(this->getInitialScheduler(), sccRowGroups);
+                    // As we removed the entries where the choice was fixed, we need to change the scheduler.
+                    // We set the scheduler to 0 for those states.
+                    if (this->choiceFixedForRowGroup) {
+                        for (auto const choice : this->choiceFixedForRowGroup.get()) {
+                            sccInitChoices[choice] = 0;
+                        }
+                    }
```

Comment:
Use `storm::utility::vector::setVectorValues` ?

Reply:
And I think here is the typo I mentioned before. It should refer to `choiceFixedForStateSCC`, right?